File: Python/KthLargestElementinaStream.py
class KthLargest(object):
    def __init__(self, k, nums):
        self.k = k
        nums.sort()
        nums = nums[-k:]
        self.nums = nums

    def add(self, val):
        if len(self.nums) < self.k:
            self.nums.append(val)
        elif val > self.nums[0]:
            self.nums[0] = val
        else:
            return self.nums[0]

        self.nums.sort()
        return self.nums[0]

File: C++/KthLargestElementinaStream.cpp
class KthLargest {
public:
    int kth;
    priority_queue <int, vector<int>, greater<int>> pq; 

    KthLargest(int k, vector<int>& nums) {
        kth = k;
        
        for (int i = 0; i<nums.size(); i++){
            add(nums.at(i));
        }
    }
    
    int add(int val) {
        pq.push(val);
        
        if (pq.size()>kth){
            pq.pop();
        }

        return pq.top();
    }
};
